.github/workflows/ci.yml

```diff
@@ -26,11 +26,11 @@ jobs:
       run: sudo apt-get update && sudo apt-get install -y fuse3 libfuse-dev
     - name: Build
       run: |
-        go build ./...
+        CGO_ENABLED=0 go build ./...
```

Comment:
what does this mean: https://tip.golang.org/doc/go1.20#cgo

isn't it unset by default

Reply:
It looks like it's enabled by default.

```
$ docker run -it golang:1.20.5 sh -c "go env | grep CGO_ENABLED"
CGO_ENABLED="1"

$ docker run -it golang:1.20 sh -c "go env | grep CGO_ENABLED"
CGO_ENABLED="1"
```

The 1.20 release notes say

```
The go command now disables cgo by default on systems without a C toolchain. More specifically, when the CGO_ENABLED environment variable is unset, the CC environment variable is unset, and the default C compiler (typically clang or gcc) is not found in the path, CGO_ENABLED defaults to 0.
```

It means it'll be unset or be 0 if a C toolchain is not present, but nothing about when it's present.

I checked for clang/gcc on fresh ubuntu/debian containers

```
$ docker run -it debian:11 sh -c "command -v clang || echo clang exists ; command -v gcc || echo gcc exists"
clang exists
gcc exists
```

I got the same results for debian:12, ubuntu20.04, debian22.04 that I checked, which indicates that the C toolchain is available on all these OSes. Haven't checked on RHEL or CentOS, but anyway it's safer to explicitly set it.

Reply:
Sorry I lost the original long reply in an accidental deletion. Short story is it is set by default.

```
gargnitin@gargnitin:~/work/cloud/storage/client/gcsfuse/tasks/202307-golang1.20.5/solutions/nocgo/proj1$ docker run -it golang:1.20.5 sh -c "go env | grep CGO_ENABLED"
CGO_ENABLED="1"
^[[Agargnitin@gargnitin:~/work/cloud/storage/client/gcsfuse/tasks/202307-golang1.20.5/solutions/nocgo/proj1$ docker run -it golang:1.20 sh -c "go env | grep CGO_ENABLED"
CGO_ENABLED="1"
```